src/src/uservardlgs.cpp

```diff
@@ -0,0 +1,538 @@
+
+void UserVarManagerGUI::DisplayInfoWindow(const wxString &title, const wxString &msg)
+{
+    InfoWindow::Display(title, msg , 8000, 1000);
+}
+
+void UserVarManagerGUI::OpenEditWindow(const wxArrayString &var)
+{
+    UsrGlblMgrEditDialog d;
+    for (unsigned int i = 0; i < var.GetCount(); ++i)
+    {
+        d.AddVar(var[i]);
+    }
+    PlaceWindow(&d);
+    d.ShowModal();
+}
+
+wxString UserVarManagerGUI::GetVariable(wxWindow* parent, const wxString &old)
+{
+    GetUserVariableDialog dlg(parent, old);
+    PlaceWindow(&dlg);
+    dlg.ShowModal();
+    return dlg.GetVariable();
+}
+
+
+BEGIN_EVENT_TABLE(GetUserVariableDialog, wxScrollingDialog)
+    EVT_BUTTON(XRCID("ID_CONFIG"), GetUserVariableDialog::OnConfig)
+    EVT_BUTTON(XRCID("wxID_OK"), GetUserVariableDialog::OnOK)
+    EVT_BUTTON(XRCID("wxID_CANCEL"), GetUserVariableDialog::OnCancel)
+    EVT_TREE_ITEM_ACTIVATED(XRCID("ID_GET_USER_VAR_TREE"), GetUserVariableDialog::OnActivated)
+END_EVENT_TABLE()
+
+GetUserVariableDialog::GetUserVariableDialog(wxWindow *parent, const wxString &old) :
+    m_old(old)
+{
+    wxXmlResource::Get()->LoadObject(this, parent, wxT("dlgGetGlobalUsrVar"), wxT("wxScrollingDialog"));
+    m_treectrl = XRCCTRL(*this, "ID_GET_USER_VAR_TREE", wxTreeCtrl);
+
+    if (m_treectrl == nullptr)
+        Manager::Get()->GetLogManager()->LogError(_("Failed to load dlgGetGlobalUsrVar"));
+
+    Load();
+
+    // Try to open the old variable
+    if (m_old != wxEmptyString && m_old.StartsWith(wxT("$(#")))
+    {
+        // Remove "$(#"
+        wxString tmp = m_old.AfterFirst('#');
+        // Remove the last ")"
+        tmp = tmp.BeforeFirst(')');
+        // In tmp is now "var.subVar". subVar is optional
+        wxString var[2];
+        var[0] = tmp.Before('.');
+        var[1] = tmp.After('.');
+        wxTreeItemId root = m_treectrl->GetRootItem();
+        wxTreeItemIdValue cookie;
+        wxTreeItemId child = m_treectrl->GetFirstChild(root, cookie);
+        unsigned int i = 0;
+        while (child.IsOk())
+        {
+            if (m_treectrl->GetItemText(child) == var[i])
+            {
+                m_treectrl->EnsureVisible(child);
+                m_treectrl->SelectItem(child);
+                i++;
+                if (i >= 2 || var[i] == wxEmptyString)
+                    break;
+
+                root = child;
+                child = m_treectrl->GetFirstChild(root, cookie);
+            }
+            else
+                child = m_treectrl->GetNextChild(root, cookie);
+        }
+    }
+
+    Fit();
+    SetMinSize(GetSize());
+}
+
+void GetUserVariableDialog::Load()
+{
+    if (m_treectrl == nullptr)
+        return;
+
+    m_treectrl->DeleteAllItems();
+
+    ConfigManager* CfgMan = Manager::Get()->GetConfigManager(wxT("gcv"));
+    const wxString &ActiveSet = Manager::Get()->GetConfigManager(wxT("gcv"))->Read(wxT("/active"));
+    wxArrayString vars = CfgMan->EnumerateSubPaths(UserVariableManager::cSets + ActiveSet + wxT("/"));
```

Comment:
This is an implementation detail of the UserVarManager, so it is something that shouldn't escape its implementation, but should be hidden behind an interface. But I suppose it is not the time to do it.